File: 4.c
#include <stdio.h>
#include <math.h>

int main()
{
    int startx, starty, endx, endy;
    scanf("%d %d", &startx, &starty);
    scanf("%d %d", &endx, &endy);

    int parking_time = (endx - startx) * 60 + endy - starty;
    int cost = 0;
    if (parking_time > 4 * 60)
    {
        cost += floor((parking_time - 4 * 60) / 30) * 60;
        parking_time = 4 * 60;
    }
    if (parking_time > 2 * 60)
    {
        cost += floor((parking_time - 2 * 60) / 30) * 40;
        parking_time = 2 * 60;
    }
    if (parking_time <= 2 * 60)
        cost += floor(parking_time / 30) * 30;
    printf("%d\n", cost);
}

File: 4.py
import math

start_time = [int(x) for x in input().split()]
end_time = [int(x) for x in input().split()]

parking_time = (end_time[0]-start_time[0])*60 + end_time[1]-start_time[1]

cost = 0
if parking_time>4*60:
    cost+=math.floor((parking_time-4*60)/30)*60
    parking_time = 4*60
if parking_time>2*60:
    cost+=math.floor((parking_time-2*60)/30)*40
    parking_time = 2*60
if parking_time<=2*60:
    cost+=math.floor(parking_time/30)*30

print(cost)